pgmpy/readwrite/ProbModelXML.py

```diff
@@ -518,6 +536,14 @@ def add_probnet_additionalconstraints(self, constraint):
             argument_value = argument.attrib['value']
             self.probnet['AdditionalConstraints'][constraint_name][argument_name] = argument_value
 
+    def add_criterion(self, criterion):
+        criterion_name = criterion.attrib['name']
+        self.probnet['DecisionCriteria'][criterion_name] = {}
+        if criterion.find('AdditionalProperties/Property') is not None:
+            for prop in criterion.findall('AdditionalProperties/Property'):
+                self.probnet['DecisionCriteria'][criterion_name]['AdditionalProperties'][prop.attrib['name']] = \
```

Comment:
There is a lot of indexing in a simple statement. Try avoiding it. Instead try creating multiple variables. This seems to be too obscure and difficult to read.